nixos/modules/services/monitoring/zabbix-proxy.nix

```diff
@@ -0,0 +1,133 @@
+# Zabbix proxy daemon.
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.zabbixProxy;
+
+  stateDir = "/run/zabbix";
+
+  logDir = "/var/log/zabbix";
+
+  libDir = "/var/lib/zabbix";
+
+  pidFile = "${stateDir}/zabbix_proxy.pid";
+
+  configFile = pkgs.writeText "zabbix_proxy.conf"
+    ''
+      Server = ${cfg.server}
+
+      LogFile = ${logDir}/zabbix_proxy
+
+      PidFile = ${pidFile}
+
+      ${optionalString (cfg.dbServer != "localhost") ''
+        DBHost = ${cfg.dbServer}
+      ''}
+
+      DBName = zabbix
+
+      DBUser = zabbix
+
+      ${optionalString (cfg.dbPassword != "") ''
+        DBPassword = ${cfg.dbPassword}
+      ''}
+
+      ${config.services.zabbixProxy.extraConfig}
+    '';
+
+  useLocalPostgres = cfg.dbServer == "localhost" || cfg.dbServer == "";
+
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.zabbixProxy.enable = mkOption {
+      default = false;
+      type = types.bool;
+      description = ''
+        Whether to run the Zabbix proxy on this machine.
+      '';
+    };
+
+    services.zabbixProxy.server = mkOption {
+       default = "127.0.0.1";
+       description = ''
+         The IP address or hostname of the Zabbix server to connect to.
+       '';
+    };
+
+    services.zabbixProxy.dbServer = mkOption {
```

Comment:
My own opinion here... feel free to take it or leave it.

In https://github.com/NixOS/nixpkgs/issues/62073 I'm trying to standardize database naming conventions and options available. It would be awesome if in this PR you added a few more options like `type`, `port`, `user`, `socket`, and `createLocally`. You could take a look at https://github.com/NixOS/nixpkgs/commit/615f8b8982b26bbb1a3e202be020d27a9f205c62 to see a decent example of what I'm thinking of.

Again... my opinion, take it or leave it. :smile: